Mage/src/main/java/mage/game/permanent/token/DragonMenaceAndStealArtifactToken.java

```diff
@@ -0,0 +1,81 @@
+package mage.game.permanent.token;
+
+import mage.MageInt;
+import mage.abilities.TriggeredAbilityImpl;
+import mage.abilities.effects.common.continuous.GainControlTargetEffect;
+import mage.abilities.keyword.FlyingAbility;
+import mage.abilities.keyword.MenaceAbility;
+import mage.constants.CardType;
+import mage.constants.Duration;
+import mage.constants.SubType;
+import mage.constants.Zone;
+import mage.filter.common.FilterArtifactPermanent;
+import mage.filter.predicate.permanent.ControllerIdPredicate;
+import mage.game.Game;
+import mage.game.events.DamagedPlayerEvent;
+import mage.game.events.GameEvent;
+import mage.players.Player;
+import mage.target.common.TargetArtifactPermanent;
+
+public class DragonMenaceAndStealArtifactToken extends TokenImpl {
+
+    public DragonMenaceAndStealArtifactToken() {
+        super("Dragon Token", "6/6 black and red Dragon creature token with flying, menace, and \"Whenever this creature deals combat damage to a player, gain control of target artifact that player controls.\"");
+        cardType.add(CardType.CREATURE);
+        color.setBlack(true);
+        color.setRed(true);
+        subtype.add(SubType.DRAGON);
+        power = new MageInt(6);
+        toughness = new MageInt(6);
+        addAbility(FlyingAbility.getInstance());
+        addAbility(new MenaceAbility(false));
+
+        addAbility(new TokenTriggeredAbility());
+    }
+
+    public DragonMenaceAndStealArtifactToken(final DragonMenaceAndStealArtifactToken token) { super(token); }
+
+    public DragonMenaceAndStealArtifactToken copy() { return new DragonMenaceAndStealArtifactToken(this); }
+
+}
+
+class TokenTriggeredAbility extends TriggeredAbilityImpl {
```

Comment:
can this please be a less generic class name? Even DragonTokenTriggeredAbility would be much better.